File: python/program/Mahasiswa.py
from SivitasAkademik import SivitasAkademik


class Mahasiswa(SivitasAkademik):
    def __init__(self,  NIM="",  fakultas="", prodi=""):  # constractor

        self.NIM = NIM
        self.fakultas = fakultas
        self.prodi = prodi
# setter getter

    def set_NIM(self, NIM):
        self.NIM = NIM

    def set_fakultas(self, fakultas):
        self.fakultas = fakultas

    def set_prodi(self, prodi):
        self.prodi = prodi

    def get_Nim(self):
        return self.NIM

    def get_fakultas(self):
        return self.fakultas

    def get_prodi(self):
        return self.prodi

File: CPP/PROGRAM/Mahasiswa.cpp
#include "SivitasAkademik.cpp"
class Mahasiswa : public SivitasAkademik
{
private: // private atribut
    string NIM;
    string prodi;
    string fakultas;

public:
    Mahasiswa(string NIK, string NIM, string nama, string jenis_kelamin, string asal_universitas, string fakultas, string email_edu) // CONSTRUCTOR
    {
        // set atribut dari Human
        setNIK(NIK);
        setNama(nama);
        setJenis_kelamin(jenis_kelamin);

        // set atribut dari SivitasAkademik
        setAsal_Universitas(asal_universitas);
        setEmail_edu(email_edu);

        // set atribut dari Mahasiswa
        this->NIM = NIM;
        this->fakultas = fakultas;
    }

    void setNIM(string b) // SET NIM
    {
        NIM = b;
    }

    void setFakultas(string d) // SET FAKULTAS
    {
        fakultas = d;
    }

    string getNIM() // GET NIM
    {
        return NIM;
    }

    string getFakultas() // GET FAKULTAS
    {
        return fakultas;
    }
    ~Mahasiswa() // DESTRUCTOR
    {
    }
};